openpype/hosts/nuke/api/lib.py

```diff
@@ -3423,3 +3423,55 @@ def create_viewer_profile_string(viewer, display=None, path_like=False):
     if path_like:
         return "{}/{}".format(display, viewer)
     return "{} ({})".format(viewer, display)
+
+
+def get_head_filename_without_hashes(original_path, name):
+    """Function to get the renamed head filename without frame hashes
+    To avoid the system being confused on finding the filename with
+    frame hashes if the head of the filename has the hashed symbol
+
+    Examples:
+        >>> get_head_filename_without_hashes("render.####.exr", "baking")
+        render.baking.####.exr
+        >>> get_head_filename_without_hashes("render.%04d.exr", "tag")
+        render.tag.%d.exr
+        >>> get_head_filename_without_hashes("exr.####.exr", "foo")
+        exr.foo.%04d.exr
+
+    Args:
+        original_path (str): the filename with frame hashes
+        name (str): the name of the tags
+
+    Returns:
+        str: the renamed filename with the tag
+    """
+    filename = os.path.basename(original_path)
+
+    def insert_name(matchobj):
+        return "{}.{}".format(name, matchobj.group(0))
+
+    return re.sub(r"(%\d*d)|#+", insert_name, filename)
```

Comment:
I saw this get merged just now but some of the discussions didn't seem entirely resolved. For example, the function name here `get_head_filename_without_hashes` AND the top description of the docstring are entirely incorrect.

It does **not** return the head of a filename, it returns a file filename.
It does **not** remove the hashes from the filename.
It does **not** avoid issues with finding files on the file system - right? How does it do that and why?

The only thing this function does is prepend **{name}.** before any frame hash (where frame hash is %d, %04d `####` or alike.

Can we get a new PR to still clean this up? :)

Reply:
@moonyuet shell we do the additional changes in extra (linked to this one please). Also include the other comment https://github.com/ynput/OpenPype/pull/5623#discussion_r1341521543